python-avd/pyavd/_cv/client/tag.py

```diff
@@ -119,19 +119,19 @@ async def get_tags(
 
         return tags
 
-    @GRPCRequestHandler()
+    @GRPCRequestHandler(list_field="tags")
     async def set_tags(
         self: CVClientProtocol,
         workspace_id: str,
-        tags: set[CVTag],
+        tags: list[CVTag],
```

Comment:
For a future PR maybe we should think about having a broader type here and cast as a list in the decorator instead. This would provide more flexibility in the calling code (we wouldn't have to convert to a list).
```suggestion
        tags: Iterable[CVTag],
```

Reply:
since we need to know the length to be able to split, I think we need to use `Sequence[CVTag]`. Also an iteratable could be consumed after the first iteration worst case.

Reply:
Agreed Sequence is a better fit here.

Reply:
Please check my latest updates, i rolled back this changes (so async functions are using specific typing like list/set). I tried to modify async decorator to handle all functions (with `list`, `set` or `tuple` being a type of the parameter that needs to be splitted). This gives a flexibility of supporting different functions with different types of iterables at the same time. Added/adjusted tests as well to confirm functionality